### DISCUSSION TITLE
Preview functionality on new page is inconsistent

### DISCUSSION THREAD

**Discussion by richardfrankza** [craftcms/cms#9017] [Ideas]
### Description

When creating a new channel entry, Gatsby Cloud Preview takes approximately 5 seconds to build my page. During that time if I click preview or "view" it returns a 404 error (nginx plain 404). It appears I need to fully refresh the page in order to see the preview as the iframe does not refresh automatically.

1. It would be great if we could have a refresh button built into the preview section to refresh the iframe manually without having to use the browser refresh.

2. Is there any way we can resolve the temporary 404 - I would hope the preview server would wait a few seconds before returning a 404 - I wouldn't mind it "pending" as the page is built. This may be a Gatsby Cloud specific request.

**Comment by andris-sevcenko** [comment]
The 1st issue you bring up is something that should probably happen in Craft CMS, not the helper plugin, so I'm transferring this issue.

The 2nd issue would be something for Gatsby Cloud to implement, not Craft.
